In [4]:
import pandas as pd

In [5]:
import numpy as np

In [13]:
df = pd.DataFrame(data = {
    "quantity": [10, 15, 12, 8, 9, 11, 14, 13, 10, 12],
    "quality": [8, 9, 7, 6, 8, 7, 9, 8, 7, 8],
    "cost": [100, 120, 110, 90, 95, 105, 115, 110, 100, 105]
}
)

In [14]:
df

quantity  quality  cost
0        10        8   100
1        15        9   120
2        12        7   110
3         8        6    90
4         9        8    95
5        11        7   105
6        14        9   115
7        13        8   110
8        10        7   100
9        12        8   105

## Finding average of all features

In [22]:
df['quantity'].mean()

11.4

###### On average, the quantity of material produced is around 11.4 units.

In [23]:
df['quality'].mean()

7.7

#### The average quality rating stands at 7.7

In [25]:
df['cost'].mean()

105.0

#### On average, it costs 105.0 to make the material.

### Finding the most common value and the next most common value in quantity

In [31]:
value_count = df['quantity'].value_counts()

In [38]:
difference = value_count.index[1] - value_count.index[0]  
difference

2

#### The most common quantity produced is 10, occurring 2 times. The next most occurring quantity is 12, also appearing 2 times. The difference between the most common quantity (10) and the next most common quantity (12) is 2 units